Sources/WFOAuth2/WFOAuth2SessionManager.m

```diff
@@ -256,8 +256,15 @@ - (WKWebView *)authorizationWebViewWithURL:(NSURL *)authorizationURL
         if ([redirectComponents.path hasSuffix:@"/"])
             redirectComponents.path = [redirectComponents.path substringToIndex:(redirectComponents.path.length - 1)];
         
+        /*
+            Square requires you to save the redirect URL in the webportal, there is no sense in passing it in as a parameter.
+            The assumption is made that redirecting to something other than the square domain is considered redirecting to your redirect url.
+         */
+        if (!redirectURI && [components.host containsString:@"squareup"])
```

Comment:
> Square requires you to save the redirect URL in the webportal, there is no sense in passing it in as a parameter.

We pass it as a parameter for two reasons:
1. Many services (not Square, it seems) use it to verify if a request is valid
2. It helps us identify which navigation actions signify completion

Unless there is a way to deal with `2` in a clean way I am hesitant to merge this as is. I'd consider a clean way to be something like this:
1. Containing this peculiarity to the Square subclass somehow
2. Recognizing completion for a `nil` `redirectURI` in a more generic way (determining if the host of the website changes, or scanning all URLs for the completion parameters)